pkg/trace/stats/concentrator_test.go

```diff
@@ -227,59 +263,70 @@ func TestConcentratorStatsCounts(t *testing.T) {
 	// Build a trace with stats which should cover 3 time buckets.
 	trace := pb.Trace{
 		// more than 2 buckets old, should be added to the 2 bucket-old, first flush.
-		testSpan(1, 0, 111, 10, "A1", "resource1", 0),
-		testSpan(1, 0, 222, 3, "A1", "resource1", 0),
+		testSpan(1, 0, 111, 10, "query", "A1", "resource1", 0, nil),
+		testSpan(1, 0, 222, 3, "query", "A1", "resource1", 0, nil),
 		// 2 buckets old, part of the first flush
-		testSpan(1, 0, 24, 2, "A1", "resource1", 0),
-		testSpan(2, 0, 12, 2, "A1", "resource1", 2),
-		testSpan(3, 0, 40, 2, "A2", "resource2", 2),
-		testSpan(4, 0, 300000000000, 2, "A2", "resource2", 2), // 5 minutes trace
-		testSpan(5, 0, 30, 2, "A2", "resourcefoo", 0),
+		testSpan(1, 0, 24, 2, "query", "A1", "resource1", 0, nil),
+		testSpan(2, 0, 12, 2, "query", "A1", "resource1", 2, nil),
+		testSpan(3, 0, 40, 2, "query", "A2", "resource2", 2, nil),
+		testSpan(4, 0, 300000000000, 2, "query", "A2", "resource2", 2, nil), // 5 minutes trace
+		testSpan(5, 0, 30, 2, "query", "A2", "resourcefoo", 0, nil),
 		// 1 bucket old, part of the second flush
-		testSpan(6, 0, 24, 1, "A1", "resource2", 0),
-		testSpan(7, 0, 12, 1, "A1", "resource1", 2),
-		testSpan(8, 0, 40, 1, "A2", "resource1", 2),
-		testSpan(9, 0, 30, 1, "A2", "resource2", 2),
-		testSpan(10, 0, 3600000000000, 1, "A2", "resourcefoo", 0), // 1 hour trace
+		testSpan(6, 0, 24, 1, "query", "A1", "resource2", 0, nil),
+		testSpan(7, 0, 12, 1, "query", "A1", "resource1", 2, nil),
+		testSpan(8, 0, 40, 1, "query", "A2", "resource1", 2, nil),
+		testSpan(9, 0, 30, 1, "query", "A2", "resource2", 2, nil),
+		testSpan(10, 0, 3600000000000, 1, "query", "A2", "resourcefoo", 0, nil), // 1 hour trace
```

Comment:
These are largely changes due to the change to the `testSpan` function. We have to pass the operation name and the meta map.

Reply:
Can you create a separate function e.g. `testSpanWithMeta`, so we don't have so much code churn on these tests? With any change, you want to be sure existing tests pass, validating the compatibility of the change. If you have a bunch of changes in the old tests, it lowers the level of confidence that you haven't broken something.